src/serialize.h

```diff
@@ -375,6 +378,160 @@ I ReadVarInt(Stream& is)
 #define VARINT(obj) REF(WrapVarInt(REF(obj)))
 #define LIMITED_STRING(obj,n) REF(LimitedString< n >(REF(obj)))
 
+class CMFToken
+{
+public:
+    CMFToken() : tag(0), format(PositiveNumber) {}
+    CMFToken(uint32_t tag, const uint256& inData)
+        : tag(tag),
+        format(ByteArray)
+    {
+        std::vector<char> tmp;
+        tmp.resize(32);
+        std::copy(inData.begin(), inData.end(), tmp.begin());
+        data = tmp;
+    }
+    explicit CMFToken(uint32_t tag, bool in = true) : tag(tag), format(in ? BoolTrue : BoolFalse) {}
+    CMFToken(uint32_t tag, int32_t in) : tag(tag), format(in >= 0 ? PositiveNumber : NegativeNumber), data(in) {}
+    CMFToken(uint32_t tag, uint64_t in) : tag(tag), format(PositiveNumber), data(in) {}
+    CMFToken(uint32_t tag, std::string in) : tag(tag), format(String), data(in) {}
+    CMFToken(uint32_t tag, std::vector<char> in) : tag(tag), format(ByteArray), data(in) {}
+    typedef boost::variant<int32_t, bool, uint64_t, std::string, std::vector<char>, double> variant;
+    enum Format {
+        PositiveNumber = 0,
+        NegativeNumber,
+        String,
+        ByteArray,
+        BoolTrue,
+        BoolFalse,
+        Double
+    };
+    uint32_t tag;
+    Format format;
+    variant data;
+
+    int intData() {
+        if (data.which() == 0)
+            return boost::get<int32_t>(data);
+        else if (data.which() == 2)
+            return boost::get<uint64_t>(data);
+        return 0;
+    }
+    uint64_t longData() {
+        if (data.which() == 0)
+            return boost::get<int32_t>(data);
+        else if (data.which() == 2)
+            return boost::get<uint64_t>(data);
+        return 0;
+    }
```

Comment:
I'd rather keep the knowledge of what is stored in the variant local in the class. Your suggestion would throw an exaction if a caller didn't exactly use the 2 types of ints supported here.